### PR TITLE
Docs updates (#1201)

### PR DESCRIPTION
**Pull request by avillela**
Changes per #1201:
* Content updates to `_index.md`
* Add `observability-primer.md`
* Content updates to `what-is-opentelemetry.md`

Preview:

- https://deploy-preview-1358--opentelemetry.netlify.app/docs/
- https://deploy-preview-1358--opentelemetry.netlify.app/docs/concepts/
- https://deploy-preview-1358--opentelemetry.netlify.app/docs/concepts/glossary/

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,6 +5,7 @@ resources/
 
 # npm assets
 node_modules/
+content-modules/
```
Comment (chalin) [inline .gitignore:8]: We need the `content-modules` under version control. I'm curious: is there a reason why you're wanting to ignore them?

Comment (chalin) [reply .gitignore]: Probably best to just drop the changes to the file (we don't need the distraction of an extra blank line either). Thanks!

File: content/en/docs/_index.md [hunk 2]
```diff
@@ -4,28 +4,9 @@ linkTitle: Docs
...
 ---
 
-OpenTelemetry is a set of APIs, SDKs, tooling and integrations that are designed
-for the creation and management of _telemetry data_ such as traces, metrics, and
-logs. The OpenTelemetry documentation is intended to broadly cover key terms,
-concepts, and instructions on how to use OpenTelemetry in your software.
+OpenTelemetry, also known as OTel for short, is a vendor-neutral open-source [Observability](concepts/observability-primer/#what-is-observability) framework for instrumenting, generating, collecting, and exporting telemetry data ([traces](concepts/observability-primer/#distributed-traces), [metrics](concepts/observability-primer/#reliability--metrics), [logs](concepts/observability-primer/#logs)). As an industry-standard it is [natively supported by a number of vendors](/vendors)
 
-![Implementation of the OpenTelemetry Reference Architecture](https://raw.github.com/open-telemetry/opentelemetry.io/main/iconography/Reference_Architecture.svg)
+![OpenTelemetry Reference Architecture](/img/otel_diagram.webp)
```
Comment (chalin) [inline content/en/docs/_index.md:9]: This might just be temporary, but in case it isn't, I'd like to flag that webp isn't as widely usable for iOS and macOS users. We should consider using another format (other than .webp and .tiff).

(For the .webp fineprint see, https://caniuse.com/?search=webp.)

Comment (avillela) [reply content/en/docs/_index.md:9]: Noted - I will convert those ASAP so that it doesn't fall off my radar.

**Comment by chalin** [comment]
@avillela - could you rebase? I've normalized some of the .md files that you've been working on. It seemed to me better to do this now than later when you'll have touched more files. It seemed to me that resolving merge conflicts shouldn't be too difficult at this point (since most of your significant changes are additions). Sorry for the extra work, and thanks for your understanding.

**Comment by avillela** [comment]
> @avillela - could you rebase? I've normalized some of the .md files that you've been working on. It seemed to me better to do this now than later when you'll have touched more files. It seemed to me that resolving merge conflicts shouldn't be too difficult at this point (since most of your significant changes are additions). Sorry for the extra work, and thanks for your understanding.

No sweat. Rebased. I also squashed all of my commits into a single commit on this branch.

File: content/en/docs/_index.md [hunk 3]
```diff
@@ -4,28 +4,9 @@ linkTitle: Docs
 menu: { main: { weight: 10 } }
 ---
 
-OpenTelemetry is a set of APIs, SDKs, tooling and integrations that are designed
-for the creation and management of _telemetry data_ such as traces, metrics, and
-logs. The OpenTelemetry documentation is intended to broadly cover key terms,
-concepts, and instructions on how to use OpenTelemetry in your software.
+OpenTelemetry, also known as OTel for short, is a vendor-neutral open-source [Observability](concepts/observability-primer/#what-is-observability) framework for instrumenting, generating, collecting, and exporting telemetry data ([traces](concepts/observability-primer/#distributed-traces), [metrics](concepts/observability-primer/#reliability--metrics), [logs](concepts/observability-primer/#logs)). As an industry-standard it is [natively supported by a number of vendors](/vendors)
```
Comment (svrnm) [inline content/en/docs/_index.md:7]: Can we keep the "such as" for traces, metrics, logs if we ever plan to have additional signals.

File: content/en/docs/_index.md [hunk 4]
```diff
@@ -4,28 +4,9 @@ linkTitle: Docs
...
 
-![Implementation of the OpenTelemetry Reference Architecture](https://raw.github.com/open-telemetry/opentelemetry.io/main/iconography/Reference_Architecture.svg)
+![OpenTelemetry Reference Architecture](/img/otel_diagram.png)
```
Comment (svrnm) [inline content/en/docs/_index.md:9]: a suggestion, could be solved in a separate PR: it's good practice to provide a description of the picture in the alt text so that people who don't see pictures still get some context.

File: content/en/docs/concepts/observability-primer.md [hunk 5]
```diff
@@ -0,0 +1,306 @@
+---
+title: Observability Primer
+weight: 9
+---
+
+Before digging into OpenTelemetry, it is important to understand some core concepts first.
+
+## What is Observability?
+
+Observability lets us understand a system from the outside, by letting us ask questions about that system without knowing its inner workings. Furthermore, allows us to easily troubleshoot and handle novel problems (i.e. “unknown unknowns”), and helps us answer the question, "Why is this happening?" *without changing your code*.
```
Comment (svrnm) [inline content/en/docs/concepts/observability-primer.md:10]: is this a definition ob observability everybody within the OTel community can "live with" (I can live with it, I am just highlighting that there are LOTS of definitions for that)

> *without changing your code*

For someone who is new to opentelemetry this part of the sentence might be confusing when you read the next one ("code must be properly instrumented"), maybe that "without chaning your code" needs some qualification

File: content/en/docs/concepts/observability-primer.md [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+
+In order to be able to ask those questions of a system, code must be properly instrumented. **OpenTelemetry** is the mechanism by which application code is instrumented, to help make a system observable.
```
Comment (svrnm) [inline content/en/docs/concepts/observability-primer.md:12]: Is an explanation of "instrumented" required? Can we point to the glossary?

File: content/en/docs/concepts/observability-primer.md [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+
+## Reliability & Metrics
```
Comment (svrnm) [inline content/en/docs/concepts/observability-primer.md:14]: I am just wondering if this should be forked out of this page, looks like a lot to read for a new comer.

File: content/en/docs/concepts/observability-primer.md [hunk 8]
```diff
@@ -0,0 +1,306 @@
...
+
+**SLI**, or Service Level Indicator, represents an item that is being measured. Although SLI created off of [metrics](#reliability--metrics), these metrics are derived from aggragated [Trace](#distributed-traces) data. An example SLI: number of successful HTTP requests / total HTTP requests (success rate).
+
+**SLO**, or Service Level Objective, is the means by which reliability is communicated to an organization/other teams. This is accomplished by attaching one or more SLIs to business value.
+
+## Understanding Distributed Tracing
+
+To understand Distributed Tracing, let's first build on a few concepts.
+
+### Logs
```
Comment (svrnm) [inline content/en/docs/concepts/observability-primer.md:30]: bunch of those concepts are already explained at other places, do we want to consolidate them?

File: content/en/docs/concepts/what-is-opentelemetry.md [hunk 9]
```diff
@@ -3,34 +3,42 @@ title: "What is OpenTelemetry?"
...
-telemetry data has been provided by either open-source projects or commercial
-vendors. With a lack of standardization, the net result is the lack of data
-portability and the burden on the user to maintain the instrumentation.
+## So what?
 
-The OpenTelemetry project solves these problems by providing a single,
-vendor-agnostic solution. The project has broad industry support and adoption
-from cloud providers, vendors and end users.
+In order to make a system observable, it must be instrumented. That is, the code must emit [traces](/docs/concepts/observability-primer/#distributed-traces), [metrics](/docs/concepts/observability-primer/#reliability--metrics), [logs](/docs/concepts/observability-primer/#logs). The instrumented data must then be sent to an Observability back-end. There are a number of Observability back-ends out there, ranging from self-hosted open-source tools (e.g. [Jaeger](https://www.jaegertracing.io/) and [Zipkin](https://zipkin.io/)), to commercial SAAS offerings (e.g. [Datadog](https://datadog.com), [Dynatrace](https://dynatrace.com), [Lighstep](https://lightstep.com), [Honeycomb](https://honeycomb.io), etc.).
```
Comment (svrnm) [inline content/en/docs/concepts/what-is-opentelemetry.md:14]: as far as I know we don't advertise commercial solutions, so take out datadog, dynatrace, lightstep, honeycomb.

**Review by svrnm — COMMENTED** [review]
Overall this looks good, although there are a few concepts that are already explained at other places, so it would make sense to consolidate them.

File: content/en/docs/_index.md (hunk 3 above)
Comment (avillela) [reply content/en/docs/_index.md:7]: Makes sense - will do!

Comment (mjingle) [inline content/en/docs/_index.md:7]:
```suggestion
OpenTelemetry, also known as OTel for short, is a vendor-neutral open-source [Observability](concepts/observability-primer/#what-is-observability) framework for instrumenting, generating, collecting, and exporting telemetry data ([traces](concepts/observability-primer/#distributed-traces), [metrics](concepts/observability-primer/#reliability--metrics), [logs](concepts/observability-primer/#logs)). As an industry-standard it is [natively supported by a number of vendors](/vendors).
```

File: content/en/docs/concepts/observability-primer.md [hunk 10]
```diff
@@ -0,0 +1,306 @@
...
+
+Observability lets us understand a system from the outside, by letting us ask questions about that system without knowing its inner workings. Furthermore, allows us to easily troubleshoot and handle novel problems (i.e. “unknown unknowns”), and helps us answer the question, "Why is this happening?" *without changing your code*.
+
+In order to be able to ask those questions of a system, code must be properly instrumented. **OpenTelemetry** is the mechanism by which application code is instrumented, to help make a system observable.
+
+## Reliability & Metrics
+
+**Telemetry** refers to data emitted from a system, about its behaviour. The data can come in the form of [Traces](#distributed-traces), [Metrics](#reliability--metrics), and [Logs](#logs).
+
+**Reliability** answers the question: “Is the service doing what users expect it to be doing?” A system could be up 100% of the time, but if, when a user clicks “Add to Cart” to add a black pair of pants to their shopping card, and instead, the system keeps adding a red pair of pants, then the system would be said to be ***un***reliable.
```
Comment (mjingle) [inline content/en/docs/concepts/observability-primer.md:18]:
```suggestion
**Reliability** answers the question: “Is the service doing what users expect it to be doing?” A system could be up 100% of the time, but if, when a user clicks “Add to Cart” to add a black pair of pants to their shopping card, and instead, the system keeps adding a red pair of pants, then the system would be said to be **un**reliable.
```

**Comment by chalin** [comment]
> @chalin No sweat. Rebased. I also squashed all of my commits into a single commit on this branch.

Thanks! After you're done with the latest round of updates, could you run the following command before resubmitting updates?

```console
$ npx prettier --write content/en/docs/concepts
```

File: content/en/docs/concepts/observability-primer.md (hunk 5 above)
Comment (avillela) [reply content/en/docs/concepts/observability-primer.md:10]: Good point. I've updated the text to:

>Observability lets us understand a system from the outside, by letting us ask questions about that system without knowing its inner workings. Furthermore, allows us to easily troubleshoot and handle novel problems (i.e. “unknown unknowns”), and helps us answer the question, “Why is this happening?”
>
>In order to be able to ask those questions of a system, the application must be properly instrumented. That is, the application code must emit traces, metrics, and logs. An application is properly instrumented when developers don’t need to add more instrumentation to troubleshoot an issue, because they have all of the information they need.

File: content/en/docs/concepts/observability-primer.md (hunk 6 above)
Comment (avillela) [reply content/en/docs/concepts/observability-primer.md:12]: I elaborated in the paragraph, as per above.

File: content/en/docs/concepts/observability-primer.md (hunk 7 above)
Comment (avillela) [reply content/en/docs/concepts/observability-primer.md:14]: I'm working on some changes that clean up this page a bit - I've added an "Advanced Concepts" page, which digs more deeply into Otel-specific constructs around traces and spans, so hopefully this page should look a bit cleaner. Stay tuned!

File: content/en/docs/concepts/what-is-opentelemetry.md (hunk 9 above)
Comment (avillela) [reply content/en/docs/concepts/what-is-opentelemetry.md:14]: Noted - will remove!

**Comment by avillela** [comment]
> Overall this looks good, although there are a few concepts that are already explained at other places, so it would make sense to consolidate them.

I'm thinking that we can get rid of [Data sources](content/en/docs/data-sources.md) and merge the content into the Advanced OTel Concepts page that I've created to separate out the OTel-specific span and tracing concepts.

File: content/en/docs/concepts/observability-primer.md (hunk 8 above)
Comment (avillela) [reply content/en/docs/concepts/observability-primer.md:56]: Yes, most definitely. In giving it a quick glance, I can see that we can probably consolidate the content from `Data sources` into `Advanced OTel Concepts` (newly added), and remove `Data sources`.

**Review by chalin — COMMENTED** [review]
You shouldn't be updating the spec in this PR:

> image

Might you have forgotten to rebase, or are you trying to access something in the spec that was committed after v1.11.0?

File: content/en/acknowledgements.md [hunk 11]
```diff
@@ -0,0 +1,19 @@
+---
+title: Acknowledgements
+menu: { main: { weight: 100 } }
```
Comment (chalin) [inline content/en/acknowledgements.md:3]: We don't want to add this to the top nav IMHO.

File: content/en/acknowledgements.md [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+---
+
+{{% blocks/lead color="primary" %}}
+# Acknowledgements
+{{% /blocks/lead %}}
+
+{{% blocks/section type="section" color="white" %}}
+
+
+We would like to acknowledge the following sources for some of the content on this site:
```
Comment (chalin) [inline content/en/acknowledgements.md:13]: I have a key question: have we been granted permission to use content from those sources? If not, this can be a problem since _our_ docs are CC-by-4, but the original docs might not be. I suppose that if the source docs permit copying / use as long as there's attribution, then we're ok.

Either way, the prose here should probably make that clear (i.e., that we have permission to copy and/or make use of the docs from these sources).

File: content/en/acknowledgements.md [hunk 13]
```diff
@@ -0,0 +1,19 @@
+---
+title: Acknowledgements
```
Comment (chalin) [inline content/en/acknowledgements.md:2]: Interesting idea!

Comment (chalin) [reply content/en/acknowledgements.md:2]: I would put this page under `/docs`, with a heavy `weight` so that it appears at the bottom of the left docs nav. Thoughts?

**Comment by avillela** [comment]
> You shouldn't be updating the spec in this PR:
> 
> 
> 
> > image
> 
> 
> 
> Might you have forgotten to rebase, or are you trying to access something in the spec that was committed after v1.11.0?

I had rebased just beforehand pushing the latest changes. I have no idea what happened. I'll take a look later.

File: content/en/acknowledgements.md [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+menu: { main: { weight: 100 } }
+---
+
+{{% blocks/lead color="primary" %}}
+# Acknowledgements
```
Comment (chalin) [inline content/en/acknowledgements.md:7]:
```suggestion
# {{% param title %}}
```

File: content/en/acknowledgements.md [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+{{% /blocks/lead %}}
+
+{{% blocks/section type="section" color="white" %}}
```
Comment (chalin) [inline content/en/acknowledgements.md:10]: If we move this page to under `/docs`, as was suggested earlier, then we'll need to drop the `blocks/*` shortcodes.

**Comment by avillela** [comment]
> You shouldn't be updating the spec in this PR:
> 
> 
> 
> > image
> 
> 
> 
> Might you have forgotten to rebase, or are you trying to access something in the spec that was committed after v1.11.0?

I had rebased just before pushing the latest changes. I have no idea what happened. I'll take a look later.

File: content/en/docs/concepts/glossary.md [hunk 16]
```diff
@@ -9,189 +9,369 @@ others. This page captures terminology used in the project and what it means.
...
 
 ### Traces
 
-- **[Trace API
-  Terminology]({{< relref "/docs/reference/specification/trace/api" >}})**
-- **[Trace SDK
-  Terminology]({{< relref "/docs/reference/specification/trace/sdk" >}})**
+##### **[Trace API Terminology](/docs/reference/specification/trace/api)**
+
+##### **[Trace SDK Terminology](/docs/reference/specification/trace/sdk)**
```
Comment (chalin) [inline content/en/docs/concepts/glossary.md:319]: I don't think that these sub-sub-entries under **Additional Terminology** need to be headings, in particular since they are links to other resources. I'd leave these as lists.

File: content/en/docs/concepts/glossary.md [hunk 17]
```diff
@@ -9,189 +9,369 @@ others. This page captures terminology used in the project and what it means.
...
-- **[Trace]({{< relref "/docs/reference/specification/overview#traces" >}}):** A
-  `DAG` of `Spans`, where the edges between `Spans` are defined as parent/child
-  relationship.
-- **[Tracer]({{< relref "/docs/reference/specification/trace/api#tracer" >}}):**
-  Responsible for creating `Spans`.
-- **Transaction:** See `Distributed Tracing`.
-- **[zPages][]:** An in-process alternative to external exporters. When
-  included, they collect and aggregate tracing and metrics information in the
-  background; this data is served on web pages when requested.
+### **Aggregation**
```
Comment (chalin) [inline content/en/docs/concepts/glossary.md:12]: Wow, nice update to the glossary. I don't know how long it will take before this PR is merged, but if the delay is too long, I'd pull this out into a separate PR and merge it first. Thoughts @avillela?

**Review by chalin — COMMENTED** [review]
A few more comments, see inline.

In particular, note the suggestion to submit the changes to the Glossary earlier.

I'm calling it a day. Will continue my review another day.

File: content/en/acknowledgements.md (hunk 12 above)
Comment (cartermp) [reply content/en/acknowledgements.md:13]: For the relevant sources that came from me (actually my employer), yes, there is permission to use that content. I also fully expect the content here to be further modified and integrated with other stuff with time.

File: content/en/docs/concepts/glossary.md (hunk 17 above)
Comment (cartermp) [reply content/en/docs/concepts/glossary.md:12]: Yes, this is a fantastic update but I think it might be easiest to pull it out into a separate PR. It would make reviewing much easier.

File: content/en/docs/concepts/observability-primer.md [hunk 18]
```diff
@@ -0,0 +1,128 @@
...
+Observability lets us understand a system from the outside, by letting us ask
+questions about that system without knowing its inner workings. Furthermore,
+allows us to easily troubleshoot and handle novel problems (i.e. “unknown
+unknowns”), and helps us answer the question, "Why is this happening?"
+
+In order to be able to ask those questions of a system, the application must be
+properly instrumented. That is, the application code must emit
+[traces](/docs/concepts/observability-primer/#distributed-traces),
+[metrics](/docs/concepts/observability-primer/#reliability--metrics),
+[logs](/docs/concepts/observability-primer/#logs). An application is properly
```
Comment (cartermp) [inline content/en/docs/concepts/observability-primer.md:20]:
```suggestion
properly instrumented. That is, the application code must emit signals such as
[traces](/docs/concepts/observability-primer/#distributed-traces),
[metrics](/docs/concepts/observability-primer/#reliability--metrics),
and [logs](/docs/concepts/observability-primer/#logs). An application is properly
```

File: content/en/docs/concepts/observability-primer.md [hunk 19]
```diff
@@ -0,0 +1,128 @@
...
+“Add to Cart” to add a black pair of pants to their shopping card, and instead,
+the system keeps adding a red pair of pants, then the system would be said to be
+**un**reliable.
+
+**Metrics** are aggregations over a period of time of numeric data about your
+infrastructure or application. Examples include: system error rate, CPU
+utilization, request rate for a given service.
+
+**SLI**, or Service Level Indicator, represents an item that is being measured.
+Although SLI created off of [metrics](#reliability--metrics), these metrics are
```
Comment (cartermp) [inline content/en/docs/concepts/observability-primer.md:44]: Not necessarily. For example, with Honeycomb, SLIs are a computed expression with one or more attributes on a trace span as input (technically not just a trace span, but this is the most common use case). I don't think most vendors do this today, but it's entirely possible that more will do this in the future. This starts to tread into "what observability backends do" territory and I do worry that if we start documenting that stuff we could quickly come off as favoring one backend over another.

File: content/en/docs/concepts/otel-concepts.md [hunk 20]
```diff
@@ -0,0 +1,394 @@
...
+
+Imagine you wanted to have the CustomerId appear on all your spans, but it’s
+only available on the initial API request because your Stock Check API doesn’t
+need a Customer context. This is where OpenTelemetry Baggage comes to the
+rescue.
+
+In OpenTelemetry, "Baggage" refers to contextual information that’s passed
+between spans. More specifically, it’s about passing that context between
+service boundaries. So really, it’s about pushing that context over an HTTP,
+gRPC, or a message so the other service can use it to add context to its span.
```
Comment (cartermp) [inline content/en/docs/concepts/otel-concepts.md:337]: This is a good scenario, but I found it a little confusing to read. I'd suggest a re-wording like so:

```suggestion
Imagine you want to have a `CustomerId` attribute on every span in your trace,
but it's only available in one specific service. To accomplish your goal, you can
use OpenTelemetry Baggage to propagate this value across your system.

In OpenTelemetry, "Baggage" refers to contextual information that’s passed
between spans. It's a key-value store that resides within a trace context,
making values available to any span created within that trace.
```

File: content/en/docs/concepts/otel-concepts.md [hunk 21]
```diff
@@ -0,0 +1,394 @@
...
+
+OpenTelemetry uses [Context Propagation](#context-propagation) to pass this
+concept around, and each of the different library implementations has
+propagators that will parse and make that Baggage available without you needing
+to explicitly implement it.
```
Comment (cartermp) [inline content/en/docs/concepts/otel-concepts.md:342]:
```suggestion
OpenTelemetry uses [Context Propagation](#context-propagation) to pass Baggage
around, and each of the different library implementations has
propagators that will parse and make that Baggage available without you needing
to explicitly implement it.
```

File: content/en/docs/concepts/otel-concepts.md [hunk 22]
```diff
@@ -0,0 +1,394 @@
...
+![OTel Baggage](/img/otel_baggage-2.png)
+
+### Baggage != Span attributes
+
+One important thing to note about Baggage is that it is not a subset of the
+[Span Attributes](#attributes). When you add something as Baggage, it does not
+automatically end up on the Attributes of the child system’s spans. You must
+explicitly take something out of Baggage and append it as Attributes.
+
+```javascript
```
Comment (cartermp) [inline content/en/docs/concepts/otel-concepts.md:384]:
```suggestion
```csharp
```

File: content/en/docs/concepts/otel-concepts.md [hunk 23]
```diff
@@ -0,0 +1,394 @@
...
+between spans. More specifically, it’s about passing that context between
+service boundaries. So really, it’s about pushing that context over an HTTP,
+gRPC, or a message so the other service can use it to add context to its span.
+
+OpenTelemetry uses [Context Propagation](#context-propagation) to pass this
+concept around, and each of the different library implementations has
+propagators that will parse and make that Baggage available without you needing
+to explicitly implement it.
+
+![OTel Baggage](/img/otel_bagage.png)
```
Comment (cartermp) [inline content/en/docs/concepts/otel-concepts.md:344]: This imagine doesn't seem to have anything about baggage in it?

**Review by cartermp — COMMENTED** [review]
Did a round of some content review. Overall I think this is great. I agree with @chalin that we may want to pull the glossary changes out into a separate PR. I have refrained from reviewing those changes in this review pass to see what is decided.

File: content/en/docs/concepts/observability-primer.md (hunk 19 above)
Comment (avillela) [reply content/en/docs/concepts/observability-primer.md]: Fair point. I've tweaked it to:

> **SLI**, or Service Level Indicator, represents a measurement of a service's behaviour. A good SLI measures your service from the perspective of your users. An example SLI can be the speed at which a web page loads.
